src/handlers/delete_client.rs

```diff
@@ -1,49 +1,22 @@
-use crate::handlers::ErrorLocation;
-use crate::handlers::{new_error_response, new_success_response, ErrorReason};
+use crate::error::Result;
+use crate::handlers::Response;
 use crate::state::AppState;
 use crate::stores::client::ClientStore;
 use crate::stores::notification::NotificationStore;
 use axum::extract::{Path, State};
-use axum::http::StatusCode;
-use axum::response::IntoResponse;
-use axum::Json;
-use serde_json::json;
 use std::sync::Arc;
 
 pub async fn handler(
     Path(id): Path<String>,
     State(state): State<Arc<AppState<impl ClientStore, impl NotificationStore>>>,
-) -> impl IntoResponse {
-    let exists = state.client_store.get_client(&id).await;
-    if exists.is_err() {
-        return (
-            StatusCode::INTERNAL_SERVER_ERROR,
-            Json(json!(new_error_response(vec![]))),
-        );
-    }
-
-    if exists.unwrap().is_none() {
-        return (
-            StatusCode::BAD_REQUEST,
-            Json(json!(new_error_response(vec![ErrorReason {
-                field: "client_id".to_string(),
-                description: "No client is registered with the supplied id".to_string(),
-                location: ErrorLocation::Body
-            }]))),
-        );
-    }
+) -> Result<Response> {
+    state.client_store.get_client(&id).await?; // Note (Harry): Throws a 404 when you try to delete a client that doesn't exists
```

Comment:
nitpick: as a user I find it more friendly to get a 2XX type response for this.

Reply:
Fixed